include/xtensor/xslice.hpp

```diff
@@ -206,6 +215,96 @@ namespace xt
         return xnewaxis_tag();
     }
 
+    template <class A, class B, class C>
+    struct xrange_adaptor
+    {
+        xrange_adaptor(A min, B max, C step)
+            : m_min(min), m_max(max), m_step(step)
+        {
+        }
+
+        template <class MIN = A, class MAX = B, class STEP = C>
```

Comment:
``MIN`` and ``MAX`` should be avoided as names, they can lead to conflict with potentially MIN and MAX macros (I know it is really ugly but you can't imagine the number of such horrors you can find in production code). It could be ``MINI`` and ``MAXI`` or whatever else, as long as it can't conflict with macros.